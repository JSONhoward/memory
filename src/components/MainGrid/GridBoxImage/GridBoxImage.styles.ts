import styled from 'styled-components'

export const StyledGridBoxFront = styled('div')<{show: boolean, name: string, id?: string}>`
position: absolute;
display: grid;
place-items: center;
opacity: ${props => props.show ? 1 : 0};
height: 100%;
width: 100%;
border-radius: 5px;
border: 1px solid rgba(0,0,0,.25);
background-image: linear-gradient(120deg, #fdfbfb 0%, #ebedee 100%);
`

export const ImageContainer = styled('div')`
position: absolute;
display: grid;
place-items: center;
font-size: 8vw;

@media screen and (orientation: portrait) {
    font-size: 15vw;
}
`