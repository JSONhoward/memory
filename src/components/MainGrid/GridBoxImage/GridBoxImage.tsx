import React from 'react'
import { ImageContainer, StyledGridBoxFront } from './GridBoxImage.styles'

interface Props {
    image: React.ReactNode,
    show: boolean,
    name: string,
    pairNum: number,
    handle: (event: any) => void
}

const GridBoxFront: React.FC<Props> = ({ name, handle, show, image, pairNum }) => {
    return (
        <StyledGridBoxFront id={pairNum.toString()} name={name} onClick={(e) => handle(e)} show={show}>
            <ImageContainer>
                {image}
            </ImageContainer>
        </StyledGridBoxFront>
    )
}

export default GridBoxFront
