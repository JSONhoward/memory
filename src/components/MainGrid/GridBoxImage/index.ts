import GridBoxImage from './GridBoxImage'
export default GridBoxImage