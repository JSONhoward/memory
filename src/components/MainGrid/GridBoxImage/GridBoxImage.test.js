import React from 'react'
import {create} from 'react-test-renderer'
import GridBoxImage from './'

const props = {
    image: <img src='#' alt='test' />,
    show: true,
    name: 'test',
    pairNum: 1,
    handle: jest.fn()
}

describe('GridBoxImage Tests', () => {
    test('Snapshot Test', () => {
        const tree = create(<GridBoxImage {...props} />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('Receives Props', () => {
        const component = create(<GridBoxImage {...props} />).root

        expect(component.props).toEqual(props)
    })
})