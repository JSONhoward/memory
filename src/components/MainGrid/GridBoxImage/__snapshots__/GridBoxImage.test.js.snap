// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridBoxImage Tests Snapshot Test 1`] = `
<div
  className="sc-AxjAm fDtckw"
  id="1"
  name="test"
  onClick={[Function]}
>
  <div
    className="sc-AxirZ hpqAHB"
  >
    <img
      alt="test"
      src="#"
    />
  </div>
</div>
`;
